pkg/controller/servicediscovery/privatednsnamespace/hooks.go

```diff
@@ -0,0 +1,160 @@
+package privatednsnamespace
+
+import (
+	"context"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/servicediscovery"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	cpresource "github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/servicediscovery/v1alpha1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+func (e *external) observeImpl(ctx context.Context, cpm cpresource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	ret := managed.ExternalObservation{}
```

Comment:
I think it would increase readability if a `managed.ExternalObservation` is created for every return statement. It is a bit hard to follow the state of the variable.

Reply:
I'm not convinced that it helps readability, but I've changed the code to create the external observation on each return